idea/explorer/src/jsonrpc.ts

```diff
@@ -30,6 +33,52 @@ export class JsonRpcServer extends JsonRpc(JsonRpcBase) {
       const result = await this.handleRequest(req.body);
       res.json(result);
     });
+
+    this._app.get('/api/voucher/:id', async (req, res) => {
+      const { genesis } = req.query;
+      if (!genesis) {
+        res.status(400).json({ error: 'Genesis not found in the request' });
+        return;
+      }
+
+      const voucherService = this._services.get(genesis.toString())?.voucher;
+      if (!voucherService) {
+        res.status(400).json({ error: 'Network is not supported' });
+        return;
+      }
+
+      try {
+        const voucher = await voucherService.getVoucher({ id: req.params.id, genesis: genesis.toString() });
+        res.json(voucher);
+      } catch (error) {
+        if (error instanceof VoucherNotFound) {
+          res.json(null);
+          return;
+        }
+        res.status(500).json({ error: 'Internal server error' });
+      }
+    });
+
+    this._app.post('/api/vouchers', async (req, res) => {
+      const { genesis } = req.query;
+      if (!genesis) {
+        res.status(400).json({ error: 'Genesis not found in the request' });
+        return;
+      }
+
+      const voucherService = this._services.get(genesis.toString())?.voucher;
+      if (!voucherService) {
+        res.status(400).json({ error: 'Network is not supported' });
+        return;
+      }
+
+      try {
+        const vouchers = await voucherService.getVouchers(req.body);
+        res.json(vouchers);
+      } catch (error) {
+        res.status(500).json({ error: 'Internal server error' });
+      }
```

Comment:
Don't you want to refactor this using decorators? As it's done with jsonrpc methods

Reply:
I decided that ideally REST should not be in the JSONRPC server at all and is a temporary solution